### ISSUE TITLE
CMake variable CGAL_VERSION on CGAL-4.8, 4.8.1 and 4.8.2 is 4.8.1000

### ISSUE THREAD

**Issue by VincentRouvreau** [CGAL/cgal#1559]
We are using CGAL on an external project and we wanted to detect that CGAL version were strictly greater or equl to CGAL 4.8.1 (because of a bug fix).

We are also using CMake to compile, and we figured out that CMake variable CGAL_VERSION on CGAL-4.8, 4.8.1 and 4.8.2 is always 4.8.1000.
And the same for last version CGAL-4.9, where CGAL_VERSION is 4.9.1000.

This information comes from CGALConfig.cmake, where is always set:
set(CGAL_MAJOR_VERSION    "4" )
set(CGAL_MINOR_VERSION    "8" )
set(CGAL_BUILD_VERSION    "1000" )

When I perform cmake, the version is well detected:
```
+ CGAL-4.8
== Setting paths ==
-- Build CGAL from release in directory CGAL-4.8
-- Packagenames: CGAL-4.8
== Generate version files ==
-- CGAL_MAJOR_VERSION=4
-- CGAL_MINOR_VERSION=8
-- CGAL_BUGFIX_VERSION=0
```

```
+ CGAL-4.8.2
== Setting paths ==
-- Build CGAL from release in directory CGAL-4.8.2
-- Packagenames: CGAL-4.8.2
...
== Generate version files ==
-- CGAL_MAJOR_VERSION=4
-- CGAL_MINOR_VERSION=8
-- CGAL_BUGFIX_VERSION=2
```

Regards,
Vincent Rouvreau

**Comment by lrineau** [comment]
Indeed, there is a bug. The third component of the `CGAL_VERSION` is supposed to be `CGAL_BUGFIX_VERSION`. Can you try if the following patch would suit you?

```patch
diff --git a/Installation/cmake/modules/CGALConfig_binary.cmake.in b/Installation/cmake/modules/CGALConfig_binary.cmake.in
index c03d935..7c817e6 100644
--- a/Installation/cmake/modules/CGALConfig_binary.cmake.in
+++ b/Installation/cmake/modules/CGALConfig_binary.cmake.in
@@ -74,6 +74,7 @@ set(CGAL_GRAPHICSVIEW_PACKAGE_DIR "@CGAL_GRAPHICSVIEW_PACKAGE_DIR@")
 
 set(CGAL_MAJOR_VERSION    "@CGAL_MAJOR_VERSION@" )
 set(CGAL_MINOR_VERSION    "@CGAL_MINOR_VERSION@" )
+set(CGAL_BUGFIX_VERSION   "@CGAL_BUGFIX_VERSION@" )
 set(CGAL_BUILD_VERSION    "@CGAL_BUILD_VERSION@" )
 set(CGAL_SCM_BRANCH_NAME  "@CGAL_SCM_BRANCH_NAME@")
 set(CGAL_GIT_SHA1         "@CGAL_GIT_SHA1@")
@@ -99,7 +100,7 @@ set(CGAL_LIBRARIES_DIR     "@CGAL_LIBRARIES_DIR@" )
 # If CGAL_ImageIO is built, tell if it was linked with Zlib.
 set(CGAL_ImageIO_USE_ZLIB                 "@CGAL_ImageIO_USE_ZLIB@" )
 
-set(CGAL_VERSION "${CGAL_MAJOR_VERSION}.${CGAL_MINOR_VERSION}.${CGAL_BUILD_VERSION}")
+set(CGAL_VERSION "${CGAL_MAJOR_VERSION}.${CGAL_MINOR_VERSION}.${CGAL_BUGFIX_VERSION}")
 
 set(CGAL_USE_FILE "${CGAL_MODULES_DIR}/UseCGAL.cmake" )
 

```

Our CGAL-4.8.x branch is already fixed, after the publication of CGAL-4.9 and CGAL-4.8.2. I can fix it in CGAL-4.9.1 but not in CGAL-4.8.x.

What is the use-case? Your users install whatever version of CGAL and you want to be compatible?

Which bug are you talking about. Maybe you could try to detect it with a call to a CMake command `try_run` with a special .cpp file.

**Comment by VincentRouvreau** [comment]
I can understand that you cannot fix it for 4.8.X and 4.9.
It is ok if it is fixed in 4.9.1.
I have tested your fix, and it works fine for me. Thank you for your reactivity.

That's it. The users install whatever version of CGAL, but all of my module will not compile in function of the version installed.

The bug I want to track is https://github.com/CGAL/cgal/issues/953
One of my module requires this bug fix. I wanted CMake to detect that CGAL version was more than 4.8.1, but that is impossible to me with this bug. I will check that CGAL version is more than 4.9.0.
Thank you for the `try_run` CMake trick.

**Comment by lrineau** [comment]
For the bug #953, if "${CGAL_MAJOR_VERSION}.${CGAL_MINOR_VERSION}" equals "4.8" then you could search if a line of `include/CGAL/Kd_tree_node.h` matches `lower_low_val`.

**Comment by lrineau** [comment]
Hi Vincent. The bug has been fixed in what will be CGA-4.9.1. By the PR #1560.

**Comment by VincentRouvreau** [comment]
Thank you !
